src/Rings/AlgClosureFp.jl

```diff
@@ -44,6 +44,8 @@ end
 elem_type(::Type{AlgClosure{T}}) where T = AlgClosureElem{T}
 parent_type(::Type{AlgClosureElem{T}}) where T = AlgClosure{T}
 
+Oscar.canonical_unit(a::AlgClosureElem) = a
```

Comment:
```suggestion
Oscar.canonical_unit(a::AlgClosureElem) = is_zero(a) ? one(a) : a
```
In fields with an easy iszero check we should use that to make sure to return a unit.

Reply:
I just went with
```
julia> canonical_unit(QQ(0))
0
```

Reply:
There are multiple issues in AA/Nemo about that.
But returning the "unit" 0 can lead to division by zero easily (in not so well-written implementations)

Reply:
Yes. It should be renamed and redesigned properly.